docs/Rootless.md

```diff
@@ -56,9 +71,24 @@ Theos supports building for the rootless scheme in a few ways:
 Additional notes:
 - You do *not* need to create separate package identifiers for rootful/rootless versions of the same package
     - Newer versions of rootless-compatible package managers (e.g., Sileo and Zebra) will present only the compatible version to users
+    - Cydia will display duplicated packages, but both lead to the rootful version
     - The behavior of other package managers varies and may or may not supply the correct package to users
 
 - All non-DEBIAN items (e.g., maintainer scripts) are placed in `/var/jb`
+    - Ensure your tweak preferences are placed in `/var/jb/var/mobile/Library/Preferences` rather than the old location of `/var/mobile/Library/Preferences`, as the latter could lead to jailbreak detection in non-jailbroken state
```

Comment:
This is no longer a problem, even for legacy preference bundle impls, with the proposed stage changes in theos/775.

Reply:
Well the bundling stage is responsible for the runtime storage location of the prefs, so that would be the point of concern. To my knowledge, rootless jailbreaks include a cfprefsd hook to redirect to the new location, which should cover the NSUserDefaults code you provide with just the suite name and possibly the CFPreferencesCopyKeyList example? Am unsure about the first NSUserDefaults example. The NSDictionary method can be solved with a simple rootless conversion macro (rootless.h), so that shouldn't pose an issue.

Reply:
> Well the bundling stage is responsible for the runtime storage location of the prefs, so that would be the point of concern.

I mean that's package/installation concern, and that should be handled, but actual prefs plist file is then created/accessed on runtime *in some place*, after package is already installed, so it has nothing to do with bundling

> The NSDictionary method can be solved with a simple rootless conversion macro (rootless.h), so that shouldn't pose an issue.

Yes, given the developer knows about it and notices it. Hence the warning, because they might miss it or not even be aware of the preferred new prefs location, given the old one should still work in such case.

It's not one of the things where you access packages files and would immediately notice the path is broken (like stuff people'd put in `/Library/Application Support`), because here you create and access a **new** file, meaning it's very easy to overlook

Reply:
`initWithSuiteName` is definitely redirected by the jailbreak (it hooks into `cfprefd`, the preferences daemon to do so) but I'm not sure about the other more elaborate patterns.

Reply:
```suggestion
    - Please ensure that any additional resources you provide (e.g., newly created or laid out files and directories) are stored in a prefixed path and not a rootful path, as the latter could lead to jailbreak detection in a non-jailbroken state
```

Reply:
Adding it to batch, but I think it should still explicitly mention `/var/mobile`. Someone could think that rootless makes tweaks unable to write in any old non-prefixed paths whatsoever previously, meanwhile /var/mobile is the only writable path that also persists into non-jb state.

And even then I'm not sure if removing explicit note of preferences is good, as it'd be the most common case of dangling files there, which someone might not immediately think of when reading the changed version...

I'd guess you don't want to necessarily mention `/var/jb` as rootless v2 or roothide are meant to not have this symlink...

Reply:
I feel like "as the latter could lead to jailbreak detection in a non-jailbroken state" covers that case as it implies that non-prefixed paths are writable but not worth writing to given the detection implications. 

Re prefs, I believe the configuration files you may be thinking of are handled by cfprefsd or preferenceloader and thus are automatically created in the desired path. I don't believe Theos provides any logic for this save for the base Root.plist, which will be laid out in the correct directory in staging. Please correct me if I am wrong though ... haven't taken a look at the preference situation in a little while.